hack/e2e/cmd/generateServicePrincipals.go

```diff
@@ -0,0 +1,218 @@
+package cmd
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	azkeyvault "github.com/Azure/azure-sdk-for-go/services/keyvault/v7.0/keyvault"
+	"github.com/Azure/go-autorest/autorest/date"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/jongio/azidext/go/azidext"
+	"github.com/spf13/cobra"
+
+	"github.com/Azure/ARO-RP/pkg/env"
+	msgraph_apps "github.com/Azure/ARO-RP/pkg/util/graph/graphsdk/applications"
+	msgraph_models "github.com/Azure/ARO-RP/pkg/util/graph/graphsdk/models"
+	"github.com/Azure/ARO-RP/pkg/util/graph/graphsdk/serviceprincipals"
+	"github.com/Azure/ARO-RP/pkg/util/keyvault"
+	utillog "github.com/Azure/ARO-RP/pkg/util/log"
+)
+
+// generateServicePrincipalsCmd represents the generateServicePrincipals command
+var generateServicePrincipalsCmd = &cobra.Command{
+	Use:   "generateServicePrincipals",
+	Short: "Generate the service principals for E2E",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		err := generateKeys(cmd)
+		if err != nil {
+			cmd.PrintErr(err)
+			return err
+		}
+		return nil
+	},
+}
+
+func init() {
```

Comment:
Sadly, Cobra forces the developer to use the init() function to register the commands/subcommands and "fun" problems can arise with the usage of init() in Go. But what can we do ...? 😆